src/coreComponents/physicsSolvers/wavePropagation/AcousticVTIWaveEquationSEM.hpp

```diff
@@ -0,0 +1,434 @@
+  {
+    static constexpr char const * sourceNodeIdsString() { return "sourceNodeIds"; }
+    static constexpr char const * sourceConstantsString() { return "sourceConstants"; }
+    static constexpr char const * sourceIsAccessibleString() { return "sourceIsAccessible"; }
+
+    static constexpr char const * receiverNodeIdsString() { return "receiverNodeIds"; }
+    static constexpr char const * receiverConstantsString() {return "receiverConstants"; }
+    static constexpr char const * receiverIsLocalString() { return "receiverIsLocal"; }
+
+    static constexpr char const * pressureNp1AtReceiversString() { return "pressureNp1AtReceivers"; }
+
+  } waveEquationViewKeys;
+
+
+  /** internal function to the class to compute explicitStep either for backward or forward.
+   * (requires not to be private because it is called from GEOS_HOST_DEVICE method)
+   * @param time_n time at the beginning of the step
+   * @param dt the perscribed timestep
+   * @param cycleNumber the current cycle number
+   * @param domain the domain object
+   * @return return the timestep that was achieved during the step.
+   */
+  real64 explicitStepInternal( real64 const & time_n,
+                               real64 const & dt,
+                               integer const cycleNumber,
+                               DomainPartition & domain );
+
+  /**
+   * @brief (Empty but must be defined) Initialize Perfectly Matched Layer (PML) information
+   */
+  virtual void initializePML() override {}
+
+  /**
+   * @brief  (Empty but must be defined) Apply Perfectly Matched Layer (PML) to the regions defined in the geometry box from the xml
+   * @param time the time to apply the BC
+   * @param domain the partition domain
+   */
+  virtual void applyPML( [[maybe_unused]] real64 const time, [[maybe_unused]] DomainPartition & domain ) override {}
+
+protected:
+
+  virtual void postProcessInput() override final;
+
+  virtual void initializePostInitialConditionsPreSubGroups() override final;
+
+private:
+
+  /**
+   * @brief Locate sources and receivers position in the mesh elements, evaluate the basis functions at each point and save them to the
+   * corresponding elements nodes.
+   * @param mesh mesh of the computational domain
+   */
+  virtual void precomputeSourceAndReceiverTerm( MeshLevel & mesh, arrayView1d< string const > const & regionNames ) override;
+
+  /**
+   * @brief Compute the surface Field indicators from the xml
+   * @param time the time to apply the BC TODO: a priori useless...
+   * @param domain the partition domain
+   */
+  virtual void precomputeSurfaceFieldIndicator( real64 const time, DomainPartition & domain );
+
+  /**
+   * @brief Apply free surface condition to the face define in the geometry box from the xml
+   * @param time the time to apply the BC
+   * @param domain the partition domain
+   */
+  virtual void applyFreeSurfaceBC( real64 const time, DomainPartition & domain ) override;
+
```

Comment:
Same as in the cpp file, you can remove the getNumNodesPerElem function and all the attributes except m_pressureNp1AtReceivers

Also it would be better to put the fields inside the WaveSolverBaseFields.hpp file, at the end all our fields will be inside it so it is better with new solver to put fields there

Reply:
I can move the Fields to `WaveSolverBaseFields.hpp`, no problem. I just copy the `AcousticWaveEquationSEM` solver (for which the fields are currently defined in [`AcousticWaveEquationSEM.hpp`](https://github.com/GEOS-DEV/GEOS/blob/develop/src/coreComponents/physicsSolvers/wavePropagation/AcousticWaveEquationSEM.hpp))